### PR TITLE
Search archives in multiple feeds

### PR DESCRIPTION
**Pull request by bekir-ozturk**
Fixes https://github.com/dotnet/install-scripts/issues/229

### CODE REVIEW

File: src/dotnet-install.ps1 [hunk 1]
```diff
@@ -66,11 +66,13 @@
     Displays diagnostics information.
 .PARAMETER AzureFeed
     Default: https://dotnetcli.azureedge.net/dotnet
-    This parameter typically is not changed by the user.
-    It allows changing the URL for the Azure feed used by this installer.
+    For internal use only.
```
Comment (vlada-shubina) [inline src/dotnet-install.ps1:69]: I would prefer to deprecate and use single add source param, but let's discuss

File: src/dotnet-install.ps1 [hunk 2]
```diff
@@ -930,25 +975,83 @@ function Get-AkaMSDownloadLink([string]$Channel, [string]$Quality, [bool]$Intern
...
 
+function Get-Feeds-To-Use()
+{
+    $feeds = @(
+    "https://dotnetcli.azureedge.net/dotnet",
+    "https://dotnetbuilds.azureedge.net/public"
+    )
+
+    if (-not [string]::IsNullOrEmpty($AzureFeed)) {
+        $feeds = @($AddSource)
```
Comment (vlada-shubina) [inline src/dotnet-install.ps1:986]: should be `$AzureFeed`?

Comment (vlada-shubina) [reply src/dotnet-install.ps1:986]: Also it should be additional to standard I believe

File: src/dotnet-install.ps1 [hunk 3]
```diff
@@ -957,7 +1060,15 @@ $NormalizedChannel = Get-NormalizedChannel $Channel
...
+$InstallRoot = Resolve-Installation-Path $InstallDir
+Say-Verbose "InstallRoot: $InstallRoot"
+$ScriptName = $MyInvocation.MyCommand.Name
+($assetName, $dotnetPackageRelativePath) = Resolve-AssetName-And-RelativePath -Runtime $Runtime
+
+$feeds = Get-Feeds-To-Use
+$DownloadLinks = @()
 
 #try to get download location from aka.ms link
 #not applicable when exact version is specified via command or json file
```
Comment (vlada-shubina) [inline src/dotnet-install.ps1:1074]: it would be good to extract this to function too

Comment (bekir-ozturk) [reply src/dotnet-install.ps1]: Done. Please see last commits.

File: src/dotnet-install.sh [hunk 4]
```diff
@@ -321,11 +320,13 @@ get_normalized_architecture_from_architecture() {
     eval $invocation
 
     local architecture="$(to_lowercase "$1")"
+
+    if [[ $architecture == \<auto\> ]]; then
```
Comment (donJoseLuis) [inline src/dotnet-install.sh:324]: minor: is case sensitivity a concern here to test for the auto case?  What if AuTO is found, for example?

**Review by donJoseLuis — APPROVED** [review]

**Comment by bekir-ozturk** [comment]
@nohwnd I'd appreciate a quick review if you have the time 🙂

**Review by nohwnd — APPROVED** [review]

**Review by DavidKarlas — APPROVED** [review]

File: src/dotnet-install.sh (hunk 4 above)
Comment (bekir-ozturk) [reply src/dotnet-install.sh:324]: I wouldn't say so. `<auto>` is our internal default value for architecture. Users are not supposed to use it. If someone tries, script will take an early exit and print a proper error message.
https://github.com/dotnet/install-scripts/blob/a1a9c0c1513d72a8f1b70a693f2ca23e72820199/src/dotnet-install.sh#L1434
https://github.com/dotnet/install-scripts/blob/a1a9c0c1513d72a8f1b70a693f2ca23e72820199/src/dotnet-install.sh#L344

Comment (ladipro) [reply src/dotnet-install.sh:324]: Also, `architecture` is actually converted to lower case just above this check.

File: src/dotnet-install.sh [hunk 5]
```diff
@@ -1119,93 +1121,219 @@ get_download_link_from_aka_ms() {
...
-    say_verbose "Normalized architecture: '$normalized_architecture'."
-    normalized_os="$(get_normalized_os "$user_defined_os")"
-    say_verbose "Normalized OS: '$normalized_os'."
-    normalized_quality="$(get_normalized_quality "$quality")"
-    say_verbose "Normalized quality: '$normalized_quality'."
-    normalized_channel="$(get_normalized_channel "$channel")"
-    say_verbose "Normalized channel: '$normalized_channel'."
-    normalized_product="$(get_normalized_product "$runtime")"
-    say_verbose "Normalized product: '$normalized_product'."
+    if [[ -n "$azure_feed" && "$azure_feed" != "" ]]; then
```
Comment (ladipro) [inline src/dotnet-install.sh:1131]: nit: The second part of the condition seems redundant. According to https://linux.die.net/man/1/test -n tests whether _the length of STRING is nonzero_.

File: src/dotnet-install.sh [hunk 6]
```diff
@@ -1119,93 +1121,219 @@ get_download_link_from_aka_ms() {
...
+        feeds=("$azure_feed")
+    fi
+
+    if [[ "$no_cdn" == "true" ]]; then
+        feeds=(
+        "https://dotnetcli.blob.core.windows.net/dotnet"
+        "https://dotnetbuilds.blob.core.windows.net/public"
+        )
+
+        if [[ -n "$uncached_feed" && "$uncached_feed" != "" ]]; then
```
Comment (ladipro) [inline src/dotnet-install.sh:1141]: Same nit with a redundant condition.

File: src/dotnet-install.sh [hunk 7]
```diff
@@ -1119,93 +1121,219 @@ get_download_link_from_aka_ms() {
...
+    fi
+
+    for link in ${download_links[@]}
+    do
+        echo "Link: $link"
+    done
+}
+
+# returns:
+#   0 - if operation succeded and the execution should continue as normal
```
Comment (ladipro) [inline src/dotnet-install.sh:1172]: nit: Typo `succeded`? Google says that both are correct, though.

**Review by ladipro — APPROVED** [review]
Looking great!

**Pull request merged**